### PR TITLE
QNTM-4876: Fix List.MaximumItem for mixed types

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Fixes Github issue: https://github.com/DynamoDS/Dynamo/issues/9009
JIRA: [QNTM-4876](https://jira.autodesk.com/browse/QNTM-4876)

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner 

### FYIs

@jnealb

### CODE REVIEW

File: src/Libraries/CoreNodes/List.cs [hunk 1]
```diff
@@ -1399,7 +1399,7 @@ private static IList IncreaseDepth(IList list, int amt)
         /// </summary>
         private static object DoubleConverter(object obj)
         {
-            if (obj is int)
+            if (obj is int || obj is long)
```
Comment (mjkkirschner) [inline src/Libraries/CoreNodes/List.cs]: is max item the only place where the double converter is used?

Comment (jnealb) [reply src/Libraries/CoreNodes/List.cs]: doubtful

Comment (ColinDayOrg) [reply src/Libraries/CoreNodes/List.cs]: Doing a quick search in the code there are only two places in List.cs where DoubleConverter appears (other than the main class definition); MinimumItem and MaximumItem

Comment (aparajit-pratap) [reply src/Libraries/CoreNodes/List.cs]: Yes

Comment (mjkkirschner) [reply src/Libraries/CoreNodes/List.cs]: yep yep, just curious if we should be looking for other regressions like this throughout the code base.

**Comment by mjkkirschner** [comment]
LGTM - thanks @aparajit-pratap

File: src/Libraries/CoreNodes/List.cs (hunk 1 above)
Comment (jnealb) [reply src/Libraries/CoreNodes/List.cs]: @mjkkirschner @aparajit-pratap Take a look at QNTM-3423 and all linked issues.

**Pull request merged**